docs/source/app-dev/app-arch.rst

```diff
@@ -293,3 +293,25 @@ As a last resort for these cases, your tests can use the reset service, which le
 The reset service has a single ``reset`` method that will cause all the accumulated state to be dropped, including all active contracts, the entire history of transactions and all allocated users. Only the DAML packages loaded in the ledger are preserved, thereby saving the time needed for reloading them as opposed to simply spinning up a new ledger.
 
 The reset service momentarily shuts down the gRPC channel it communicates over, so your testing infrastructure must take this into account and, when the ``reset`` is invoked, must ensure that tests are temporarily suspended as attempts to reconnect with the rebooted ledger are performed. There is no guarantee as to how long the reset will take, so this should also be taken into account when attempting to reconnect.
+
+.. _handling-submission-failures:
+
+Handle failures when submitting commands
+****************************************
+
+The interaction of a DAML application with the ledger is inherently asynchronous: applications send commands to the ledger, and some time later they see the effect of that command on the ledger.
+
+There are several things that can fail during this time window: the application can crash, the participant node can crash, messages can be lost on the network, or the ledger may be just slow to respond due to a high load. 
+
+If you want to make sure no command is executed twice (especially important with non-consuming choices), your application needs to robustly handle all the various failure scenarios.
+DAML ledgers provide **command deduplication** to solve this problem:
+Applications can assign a “time to live” (TTL) to each command,
```

Comment:
I would say "time to live (TTL) to each command-ID, not the command. Otherwise, people might get confused and think that we are talking here about MRT. Or you might not even want to talk about "time-to-live" but rather than command-id expiration time or so.

Reply:
Fair point. I will change that.